src/sempy_labs/_helper_functions.py

```
@@ -813,13 +1020,17 @@ def resolve_workspace_name(workspace_id: Optional[UUID] = None) -> str:
             request=f"/v1/workspaces/{workspace_id}", client="fabric_sp"
         ).json()
     except FabricHTTPException:
-        raise ValueError(
-            f"{icons.red_dot} The '{workspace_id}' workspace was not found."
-        )
+        if throw_error:
+            raise ValueError(
+                f"{icons.red_dot} The '{workspace_id}' workspace was not found."
+            )
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)